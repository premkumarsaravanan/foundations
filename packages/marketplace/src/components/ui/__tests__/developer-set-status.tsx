import React from 'react'
import { shallow } from 'enzyme'
import {
  SetDeveloperStatusModal,
  SetDeveloperStatusProps,
  onAfterCloseHandler,
  onSuccessHandler,
} from '../developer-set-status'

describe('SetDeveloperStatusModal', () => {
  it('should match snapshot', () => {
    const props = {
      developer: { id: '', isInactive: false },
      developerName: '',
      onSuccess: () => jest.fn(),
      formState: 'PENDING',
      visible: true,
      afterClose: () => jest.fn(),
      developerSetStatusRequest: () => jest.fn(),
      resetDeveloperSetStatusReducer: () => jest.fn(),
    } as SetDeveloperStatusProps

    const wrapper = shallow(<SetDeveloperStatusModal {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})

describe('onAfterCloseHandler', () => {
  it('should return a function when executing', () => {
    const mockAfterClose = jest.fn()

    const onAfterCloseHandlerFn = onAfterCloseHandler({ afterClose: mockAfterClose, isLoading: false })
    expect(onAfterCloseHandlerFn).toBeDefined()

    onAfterCloseHandlerFn()
    expect(mockAfterClose).toBeCalled()
  })
})

describe('onSuccessHandler', () => {
  it('should return a function when executing', () => {
    const mockOnSuccess = jest.fn()
    const mockResetDeveloperSetStatusReducer = jest.fn()

    const onSuccessHandlerFn = onSuccessHandler({
      onSuccess: mockOnSuccess,
      resetDeveloperSetStatusReducer: mockResetDeveloperSetStatusReducer,
    })
    expect(onSuccessHandlerFn).toBeDefined()

    onSuccessHandlerFn()
    expect(mockOnSuccess).toBeCalled()
    expect(mockResetDeveloperSetStatusReducer).toBeCalled()
  })
})
