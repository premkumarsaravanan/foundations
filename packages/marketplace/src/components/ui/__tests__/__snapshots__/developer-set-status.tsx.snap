// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetDeveloperStatusModal should match snapshot 1`] = `
<Modal
  afterClose={[Function]}
  renderChildren={true}
  visible={true}
>
  <ModalHeader
    afterClose={[Function]}
    title="Confirm undefined Deactivation"
  />
  <ModalBody
    body={
      <SubTitleH6
        isCentered={true}
      >
        Are you sure you want to 
        deactivate
         ‘
        ’?
      </SubTitleH6>
    }
  />
  <ModalFooter
    footerItems={
      <React.Fragment>
        <Button
          loading={false}
          onClick={[Function]}
          type="button"
          variant="danger"
        >
          Confirm
        </Button>
        <Button
          disabled={false}
          onClick={[Function]}
          type="button"
          variant="secondary"
        >
          Cancel
        </Button>
      </React.Fragment>
    }
  />
</Modal>
`;
